viewing_party/party.py

```diff
@@ -1,23 +1,210 @@
+
+def add_to_watchlist(user_data, movie):
+    new_movie = create_movie(movie["title"],movie["genre"],movie["rating"])
+    user_data["watchlist"] = [new_movie]
+
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:      
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum_of_ratings = 0.0
+    ratings_avg = 0.0
+    length = len(user_data["watched"])
+    if length > 0:
+        for i in range(length):
+            sum_of_ratings += user_data["watched"][i]["rating"]      
+            
+        ratings_avg = sum_of_ratings/length
 
+        return ratings_avg
+    else:
+        return 0.0
+    
+def get_most_watched_genre(user_data):
+    count_dict = {}
+    cntr = 0
+    next_genre = ""
+    length = len(user_data["watched"])
+
+    if length > 0:
+        for i in range(length):
+            next_genre += user_data["watched"][i]["genre"] 
+        
+            if next_genre in count_dict:
+                cntr += 1
+            else:
+                cntr = 1
+            
+            count_dict.update({next_genre:cntr})    
+        
+        max_value = max(count_dict, key=count_dict.get)
+
+        return max_value
+    else:
+        return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    watched_len = len(user_data["watched"])   
+    friends_len = len(user_data["friends"]) 
+    watched_title = ""
+    friends_title = ""
+    unique_movie = ""
+    unique_movies = []
+    friends_movie_titles = []
+
+    for i in range(friends_len):
+        friends_movie_num = len(user_data["friends"][i]["watched"])
+        cntr = 0
+        while cntr < friends_movie_num:
+            friends_title = user_data["friends"][i]["watched"][cntr]["title"]
+            friends_movie_titles.append(friends_title)
+            cntr += 1
+
+    for j in range(watched_len):
+        watched_title = user_data["watched"][j]["title"]
+            
+        if watched_title in friends_movie_titles:
+            continue
+        else:
+            unique_movie = user_data["watched"][j]
+            unique_movies.append(unique_movie)
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    watched_len = len(user_data["watched"])   
+    friends_len = len(user_data["friends"]) 
+    friends_title = ""
+    user_title = ""
+    unique_movie = ""
+    unique_movies = []
+    users_movie_titles = []
+
+    for i in range(watched_len):
+        user_title = user_data["watched"][i]["title"]
+        users_movie_titles.append(user_title)
+
+    for j in range(friends_len):

```

Comment:
this for loop/while loop looks similar to your code in `get_available_recs`. This would be a great candidate for a helper function.

Reply:
Good point